src/hooks/connectWeb3/useWeb3React.tsx

```diff
@@ -104,12 +104,30 @@ export function ActiveWeb3Provider({
     [jsonRpcUrlMap, defaultChainId]
   )
 
-  connections = [metaMaskConnection, walletConnectConnectionQR, walletConnectConnectionPopup, networkConnection]
-  if (integratorConnection) connections = [integratorConnection, ...connections]
+  const key = useRef(0)
+  connections = useMemo(() => {
+    // while react warns against triggering side effects in useMemo,
+    // in this instance we're only using the mutated value to generate a key,
+    // so tightly coupling the key update with the memo update shouldn't cause any issues,
+    // and most clearly expresses the intent
+    key.current += 1
+    return [
+      integratorConnection,
+      metaMaskConnection,
+      walletConnectConnectionQR,
+      walletConnectConnectionPopup,
+      networkConnection,
+    ].filter((connection): connection is Web3Connection => Boolean(connection))
```

Comment:
Nit: I think you can add the type to the template parameters and just use Boolean, so that the actual code remains simpler.

Reply:
hm i tried this and i don't think it's possible - similar issue (without a fix) described here: https://github.com/microsoft/TypeScript/issues/16069